### PR TITLE
TNO-2968: Show previous results on main screens

### PR DESCRIPTION
**Pull request by jdtoombs**
- removed `PreviousResults.tsx` from /features/filter-media to make a shared one
- made `useFetchResults` to simplify the process and eliminate duplicate code
- initially tried having all of this integrated into `ContentList.tsx` but had to revert those changes due to too many complications

`useFetchResults()` gives the developer access to `isLoading, currDateResults, prevDateResults, fetchResults (previously written on every page), and setIsLoading`

image

### CODE REVIEW

File: app/subscriber/src/features/filter-media/FilterMedia.tsx [hunk 1]
```diff
@@ -37,62 +36,14 @@ export const FilterMedia: React.FC<IFilterMediaProps> = ({
     {
       mediaType: { filter },
     },
-    { findContentWithElasticsearch, storeMediaTypeFilter: storeFilter },
+    { storeMediaTypeFilter: storeFilter },
   ] = useContent();
-  const [currDateResults, setCurrDateResults] = React.useState<IContentSearchResult[]>([]);
```
Comment (jdtoombs) [inline app/subscriber/src/features/filter-media/FilterMedia.tsx:42]: Moved to hook

File: app/subscriber/src/features/home/Home.tsx [hunk 2]
```diff
@@ -40,46 +48,12 @@ export const Home: React.FC = () => {
...
         },
       }));
-      setLoading(false);
+      setIsLoading(false);
     },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
     [filter.startDate],
   );
 
-  const fetchResults = React.useCallback(
```
Comment (jdtoombs) [inline app/subscriber/src/features/home/Home.tsx:48]: Can access this logic through new hook, also splits the results appropriately for displaying previous 5 days

File: app/subscriber/src/components/date-filter/DateFilter.tsx [hunk 3]
```diff
@@ -7,19 +7,13 @@ import { IFilterSettingsModel } from 'tno-core';
 import * as styled from './styled';
 
 export interface IDateFilterProps {
-  loaded?: boolean;
```
Comment (jdtoombs) [inline app/subscriber/src/components/date-filter/DateFilter.tsx:10]: Wasn't actually being used in any logic in this component.

File: app/subscriber/src/features/home/Home.tsx [hunk 4]
```diff
@@ -1,35 +1,43 @@
...
-  const [loading, setLoading] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!featuredStoryActionId || isLoading) return;
+    fetchResults({
+      ...filter,
+      actions: [getBooleanActionValue(featuredStoryActionId)],
+      startDate: filter.startDate ?? moment().startOf('day').toISOString(),
+    }).then(() => {});
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Fosol) [inline app/subscriber/src/features/home/Home.tsx:38]: Always add a comment explaining your decision

File: app/subscriber/src/features/top-stories/TopStories.tsx [hunk 5]
```diff
@@ -1,61 +1,46 @@
...
-      });
+      topStoryAction &&
+        fetchResults({
+          ...filter,
+          actions: [topStoryAction],
+          startDate: filter.startDate ?? moment().startOf('day').toISOString(),
+        });
     }
-  }, [filter, findContentWithElasticsearch, getActionFilters, isReady, topStoryActionId]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Fosol) [inline app/subscriber/src/features/top-stories/TopStories.tsx:42]: Always leave a comment explaining your decision

File: app/subscriber/src/features/top-stories/TopStories.tsx [hunk 6]
```diff
@@ -67,8 +52,9 @@ export const TopStories: React.FC = () => {
           selected: selectedContent,
         },
       }));
-      setLoading(false);
+      setIsLoading(false);
     },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Fosol) [inline app/subscriber/src/features/top-stories/TopStories.tsx:57]: Always leave a comment explaining your decision

Comment (jdtoombs) [reply app/subscriber/src/features/top-stories/TopStories.tsx:59]: Updated!

**Pull request merged**